File: Medium/Python/Other/EvalRPN.py
class Solution(object):
    def evalRPN(self, tokens):
        """
            :type tokens: List[str]
            :rtype: int
            """
        stack = []
        for t in tokens:
            if t in '+-*/':
                n1 = stack.pop()
                n2 = stack.pop()
                if t == '-':
                    n1 = n2 - n1
                elif t == '+':
                    n1 = n2 + n1
                elif t == '/':
                    if n2 * n1 < 0 and n2 % n1 != 0:
                        n1 = n2 / n1 + 1
                    else:
                        n1 = n2 / n1
                elif t == '*':
                    n1 = n2 * n1
                stack.append(n1)
            else:
                stack.append(int(t))
        return stack[-1]

File: Medium/Java/Others/EvalRPN.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>();
        stack.push(0);
        for (String token: tokens) {
            if (!("+-*/").contains(token))
                stack.push(Integer.valueOf(token));
            else {
                int opt2 = stack.pop(), opt1 = stack.pop();
                if (token.equals("+"))
                    opt1 += opt2;
                else if (token.equals("-"))
                    opt1 -= opt2;
                else if (token.equals("*"))
                    opt1 *= opt2;
                else if (token.equals("/"))
                    opt1 /= opt2;
                stack.push(opt1);
            }
        }
        return stack.pop();
    }
}
